src/transformers/models/qwen2/modeling_qwen2.py

```diff
@@ -203,7 +204,7 @@ def apply_rotary_pos_emb(q, k, cos, sin, position_ids=None, unsqueeze_dim=1):
     cos = cos.unsqueeze(unsqueeze_dim)
     sin = sin.unsqueeze(unsqueeze_dim)
     q_embed = (q * cos) + (rotate_half(q) * sin)
-    k_embed = (k * cos) + (rotate_half(k) * sin)
+    k_embed = (k * cos) + (rotate_half(k) * sin) if k is not None else None
```

Comment:
Minor thing, but personally I would add an alternative function that just takes a q input and returns q_embed rather than branching and handling None / returning None.

This way it's easier to reason about from the caller code because it's transparent what cases you do  / don't need to handle after the invocation occurs.